src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -99,6 +99,12 @@ def shutdown(self):
         """
         raise NotImplementedError('Abstract Method: shutdown')
 
+    def setEnv(self,name, value=None):
+        """Sets environment variables. If no value is provided it will be looked up
+        from the current environment.
+        """
+        raise NotImplementedError('Abstract method: setEnv')
```

Comment:
No need for the message argument to the exception constructor. It will be clear from the trace back which method was left unimplemented. Also, the ABC mechanism will catch unimplemented methods much earlier, at instantiation time.